### PR TITLE
logged in user can now create a new product to sell

### PR DESCRIPTION
**Pull request by colebryant**
# Description

Added controller methods and views which enable the user to enter a new product to sell.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Testing Instructions for Change Made

1. `git fetch -a`
1. `git checkout cb-products-add`
1. Run Bangazon
1. Verify that only when the user is logged in, they can access the create product view. 
1. Verify that a new product can be created, and that there is form validation for all the fields (except for city and image path). This includes the product category selection field (ticket #5 ).
1. Verify that once a new product is created, the user is redirected to the details view for that product.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [ ] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by THATBLADEBOY — APPROVED** [review]
Works as described 👍

**Review by jdbbaugh — APPROVED** [review]
Solid, may need to take out userId in the long run but works as described 🤘

**Review by dakotacookson — APPROVED** [review]
All good in the hood

File: Bangazon/Models/Product.cs [hunk 1]
```diff
@@ -30,14 +30,12 @@ public class Product
     [Required]
     public int Quantity { get; set; }
 
-    [Required]
     public string UserId {get; set;}
```
Comment (askingalot) [inline Bangazon/Models/Product.cs]: UserId should be required

File: Bangazon/Controllers/ProductsController.cs [hunk 2]
```diff
@@ -61,14 +79,16 @@ public IActionResult Create()
...
+                var user = await _userManager.GetUserAsync(HttpContext.User);
+                product.UserId = user.Id;
                 _context.Add(product);
                 await _context.SaveChangesAsync();
-                return RedirectToAction(nameof(Index));
+                return RedirectToAction("Details", new { @id = product.ProductId });
             }
-            ViewData["ProductTypeId"] = new SelectList(_context.ProductType, "ProductTypeId", "Label", product.ProductTypeId);
-            ViewData["UserId"] = new SelectList(_context.ApplicationUsers, "Id", "Id", product.UserId);
+            ViewData["ProductTypeId"] = new SelectList(_context.ProductType, "Value", "Text", product.ProductTypeId);
```
Comment (askingalot) [inline Bangazon/Controllers/ProductsController.cs]: ViewModel

File: Bangazon/Controllers/ProductsController.cs [hunk 3]
```diff
@@ -61,14 +79,16 @@ public IActionResult Create()
         [ValidateAntiForgeryToken]
         public async Task<IActionResult> Create([Bind("ProductId,DateCreated,Description,Title,Price,Quantity,UserId,City,ImagePath,ProductTypeId")] Product product)
         {
-            if (ModelState.IsValid)
+            var errors = ModelState.Values.SelectMany(v => v.Errors);
```
Comment (askingalot) [inline Bangazon/Controllers/ProductsController.cs]: What is this for?

**Review by askingalot — CHANGES_REQUESTED** [review]

**Review by askingalot — APPROVED** [review]

**Pull request merged**